### DISCUSSION TITLE
Multiline condition escaping in YAML

### DISCUSSION THREAD

**Discussion by sbalmos** [vectordotdev/vector#14916] [Q&A] [answered]
If I am writing a multiline filter condition in TOML, I can split the line for readability by appending an escaping backslash.

```
[transforms.trial]
type = "filter"
inputs = ["somewhere"]
condition = """
.message.field1 == "foo" \
  && (includes([
      "bar",
      "baz",
    ], .message.field2) \
    || .message.elsewhere_id == 3 \
  )
```
This validates and works fine. However, if I try to use that same multiline condition block in YAML format (namely so the Vector config can be placed in a customConfig block of a Vector Helm chart values file...), the condition fails to validate, with an unexpected syntax token "Escape"
```
transforms:
  trial:
    type: filter
    inputs:
    - somewhere
    condition: |
      .message.field1 == "foo" \
        && (includes([
            "bar",
            "baz",
          ], .message.field2) \
          || .message.elsewhere_id == 3 \
      )
```
I've tried various YAML multiline block formats (|, >, single-quoting strings with newlines, etc). If I remove the escape, then the && operator is unexpected by the VRL parser. Moving the && up to the previous line not just makes for bad readability IMO, but then kicks the unexpected token down to the newline, opening parenthesis, or elsewhere.

Ideas?

**Comment by neuronull** [comment, marked as answer]
Hi @sbalmos !
The VRL expressions over multiple lines is a bit arbitrary as to when it works currently- something we hope to improve going forward.
I was able to get this to pass `vector validate` with:

```
transforms:
  trial:
    type: filter
    inputs:
    - foo
    condition: |
      .message.field1 == "foo" &&
        (includes([ "bar", "baz", ], .message.field2) ||
        .message.elsewhere_id == 3)
```

**Comment by sbalmos** [comment]
Wrapping back on this... Ironically it looks like this is very dependent on exactly what YAML multiline string form you use. This works for me (note, actually, no use at all of any block-form string):

```
transforms:
  trial:
    type: filter
    inputs:
    - foo
    condition:
      .message.field1 == "foo"
        && (
            includes([
              "bar",
              "baz",
            ], .message.field2)
            || message.elsewhere_id == 3
          )
```
